Source/WebSocket.swift

```diff
@@ -1668,13 +1668,17 @@ public class WebSocket: NSObject {
     public init(request: NSURLRequest, subProtocols : [String] = []){
         opened = true
         ws = InnerWebSocket(request: request, subProtocols: subProtocols, stub: false)
+        super.init()
+        ws.eclose = { [unowned self] in
```

Comment:
Well, IMO the convenience initialiser doesn't make sense. @tidwall why do you want a `NSURLRequest` that represents an empty URL?